### PR TITLE
Add support for `arguments`-style compilation databases (#297)

### LINKED ISSUE #297: `extract_ir/convert_compile_command_to_objectfile` does not support `arguments`-style compilation databases
`extract_ir` errors on certain compilation databases. 

In this case, I generated a compilation database for redis(uses Makefile) using [bear](https://github.com/rizsotto/Bear):
```
PYTHONPATH=:/home/user/ml-opt/ml-compiler-opt python3 /home/user/ml-opt/ml-compiler-opt/compiler_opt/tools/extract_ir.py --cmd_filter=^-O2|-Os|-Oz$ --input=/home/user/ml-opt/redis_repo/compile_commands.json --input_type=json --llvm_objcopy_path=build/bin/llvm-objcopy --output_dir=/home/user/ml-opt/redis_corpus
Traceback (most recent call last):
  File "/home/user/ml-opt/ml-compiler-opt/compiler_opt/tools/extract_ir.py", line 142, in <module>
    app.run(main)
  File "/home/user/ml-opt/mlgo_venv/lib/python3.10/site-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/home/user/ml-opt/mlgo_venv/lib/python3.10/site-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "/home/user/ml-opt/ml-compiler-opt/compiler_opt/tools/extract_ir.py", line 107, in main
    objs = extract_ir_lib.load_from_compile_commands(
  File "/home/user/ml-opt/ml-compiler-opt/compiler_opt/tools/extract_ir_lib.py", line 229, in load_from_compile_commands
    objs = [
  File "/home/user/ml-opt/ml-compiler-opt/compiler_opt/tools/extract_ir_lib.py", line 230, in <listcomp>
    convert_compile_command_to_objectfile(cmd, output_dir)
  File "/home/user/ml-opt/ml-compiler-opt/compiler_opt/tools/extract_ir_lib.py", line 210, in convert_compile_command_to_objectfile
    cmd = command['command']
KeyError: 'command'
```
This is occurs because `compiler_opt/tools/extract_ir_lib.py/convert_compile_command_to_objectfile` simply checks if `command` exists, when `arguments` may be present instead(see [JSON Compilation Database Format Specification/Format](https://clang.llvm.org/docs/JSONCompilationDatabase.html#format)).

This looks very easy to resolve: if `command` is missing but `arguments` is present, convert `arguments` to `command` by concatenation.

Do you mind if I fix this and create a pull request?

### PR DESCRIPTION
**Pull request by 25pwn**
Fixes #297

### CODE REVIEW

**Comment by boomanaiden154** [comment]
Can you add test coverage for this under `compiler_opt/tools/extract_ir_test.py`?

File: compiler_opt/tools/extract_ir_lib.py [hunk 1]
```diff
@@ -207,14 +207,19 @@ def extract(self,
...
+    logging.info('compile_commands element has no command and arguments')
+    return None
 
-  cmd_parts = cmd.split()
   try:
     obj_index = cmd_parts.index('-o') + 1
   except ValueError:
     # This could happen if there are non-clang commands in compile_commands.json
-    logging.info('Command has no -o option: %s', cmd)
+    logging.info('Command has no -o option: %s', cmd_parts)
```
Comment (boomanaiden154) [inline compiler_opt/tools/extract_ir_lib.py:222]: Maybe `' '.join(cmd_parts)` to maintain the previous formatting?

**Review by boomanaiden154 — APPROVED** [review]
LGTM with one other comment that should probably be addressed.

**Comment by mtrofin** [comment]
LGTM, thanks @25pwn for doing this!

**Pull request merged**